### DISCUSSION TITLE
Vexflow 4 Migration

### DISCUSSION THREAD

**Discussion by deemaagog** [opensheetmusicdisplay/opensheetmusicdisplay#1453] [General]
Hi guys! Thank you for this awesome project! What's the current state of migrating to newer vexflow versions? I see an opened PR, but it looks abandonned?! Are there any plans to continue this work?

**Comment by sschmidTU** [comment]
It's explained pretty clearly here:
https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/pull/1139
Vexflow 3+ has layout issues, since they introduced a brand new formatter (by 0xfe), which was not the focus of improvements (though great improvements came in other parts), and shortly after 0xfe became inactive. The current best version for OSMD is 1.2.93. It produces the best results overall. It would be a lot of work to fix all the layout issues with newer Vexflow versions, and additionally adapting to all the changes, and porting our custom Vexflow patches. Also, Vexflow 5 is currently breaking a lot of older APIs, and they're not done breaking them yet, so I'd prefer to wait until most of that is finished.

**Reply by rvilarl** [reply]
@sschmidTU what are the API changes that concern you? Have you proposals? @ronyeh, myself and the rest of the team will take them into account.

**Reply by sschmidTU** [reply]
No specific concerns, it's just that the changes need to be adapted, and if we adapt them now, we have to adapt them again in 3 months or so when more APIs have changed.
So, this shouldn't be seen as a criticism of Vexflow 5, sorry if I've been a bit direct here, I don't mind the APIs being redesigned when everything is newly set up in a consistent and better way, it's just that I'd prefer waiting until the API changes are mostly done.

Also, I haven't re-checked the formatter and layout issues recently, I'm sure you've improved a few of them, so thanks for your work, and I'm looking forward to try adapting Vexflow 4 or 5 again soon, when we have the time!
It's just that it's a lot of work to switch to the new Vexflow version, especially with our custom Vexflow patches, some of which probably need to be ported, and the same principle still holds: We use the version that produces the best results overall (especially layout-/formatting-wise).
